docs/source/reference/oauth.md

```diff
@@ -0,0 +1,214 @@
+# JupyterHub and OAuth
+
+JupyterHub uses OAuth 2 internally as a mechanism for authenticating users.
+As such, JupyterHub itself always functions as an OAuth **provider**.
+More on what that means below.
+
+Additionally, JupyterHub is _often_ deployed with [oauthenticator](https://oauthenticator.readthedocs.io),
+where an external identity provider, such as GitHub or KeyCloak, is used to authenticate users.
+When this is the case, there are \*two nested
```

Comment:
Sentence fragment

Reply:
I guess it will be something like `..., there are two nested oauth flows. For example, for a user launching a server with GitHub authentication enabled, the first oauth flow to start will be the user authenticating against jupyterhub. Before completion, of that first flow, the GitHub oauth flow needs to be performed (the second flow). The detailed nested steps are detailed later on in this page`.